src/specific/s_input.c

```diff
@@ -351,61 +357,134 @@ static bool S_Input_Key(INPUT_ROLE role)
+    linput.health_cheat              = KEY_DOWN(SDL_SCANCODE_F11);
+
+    linput.equip_pistols             = KEY_DOWN(SDL_SCANCODE_1);
+    linput.equip_shotgun             = KEY_DOWN(SDL_SCANCODE_2);
+    linput.equip_magnums             = KEY_DOWN(SDL_SCANCODE_3);
+    linput.equip_uzis                = KEY_DOWN(SDL_SCANCODE_4);
+    linput.use_small_medi            = KEY_DOWN(SDL_SCANCODE_8);
+    linput.use_big_medi              = KEY_DOWN(SDL_SCANCODE_9);
+
+    linput.select                    = KEY_DOWN(SDL_SCANCODE_RETURN);
+    linput.deselect                  = S_Input_Key(INPUT_ROLE_OPTION);
+
+    linput.save                      = KEY_DOWN(SDL_SCANCODE_F5);
+    linput.load                      = KEY_DOWN(SDL_SCANCODE_F6);
+
+    linput.toggle_fps_counter        = KEY_DOWN(SDL_SCANCODE_F2);
+    linput.toggle_bilinear_filter    = KEY_DOWN(SDL_SCANCODE_F3);
     linput.toggle_perspective_filter = KEY_DOWN(SDL_SCANCODE_F4);
+    // clang-format on
+
+    if (m_Controller) {
+        // clang-format off
+        linput.forward      |= S_Input_JoyAxis(SDL_CONTROLLER_AXIS_LEFTY) < 0;
+        linput.back         |= S_Input_JoyAxis(SDL_CONTROLLER_AXIS_LEFTY) > 0;
+        linput.left         |= S_Input_JoyAxis(SDL_CONTROLLER_AXIS_LEFTX) < 0;
+        linput.right        |= S_Input_JoyAxis(SDL_CONTROLLER_AXIS_LEFTX) > 0;
+        linput.step_left    |= S_Input_JoyAxis(SDL_CONTROLLER_AXIS_TRIGGERLEFT) > 0;
+        linput.step_right   |= S_Input_JoyAxis(SDL_CONTROLLER_AXIS_TRIGGERRIGHT) > 0;
+        linput.camera_left  |= S_Input_JoyAxis(SDL_CONTROLLER_AXIS_RIGHTX) < 0;
+        linput.camera_right |= S_Input_JoyAxis(SDL_CONTROLLER_AXIS_RIGHTX) > 0;
+        linput.camera_up    |= S_Input_JoyAxis(SDL_CONTROLLER_AXIS_RIGHTY) < 0;
+        linput.camera_down  |= S_Input_JoyAxis(SDL_CONTROLLER_AXIS_RIGHTY) > 0;
+        linput.forward      |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_DPAD_UP);
+        linput.right        |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_DPAD_RIGHT);
+        linput.left         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_DPAD_LEFT);
+        linput.back         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_DPAD_DOWN);
+        linput.action       |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_A);
+        linput.select       |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_A);
+        linput.roll         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_B);
+        linput.deselect     |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_B);
+        linput.jump         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_X);
+        linput.draw         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_Y);
+        linput.look         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_LEFTSHOULDER);
+        linput.slow         |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_RIGHTSHOULDER);
+        linput.option       |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_BACK);
+        linput.deselect     |= S_Input_JoyBtn(SDL_CONTROLLER_BUTTON_TOUCHPAD);
```

Comment:
Is deselect supposed to be used twice here? Set to TOUCHPAD and B button.

Reply:
Hmm I kept it because I felt like it but I can remove it.

Reply:
Oh that's fine. Just wanted to make sure it wasn't an oversight. Makes sense to be able to close the inventory in game with the same button you opened it with.

Reply:
On that note maybe I should add this behavior to the pause and the alt inventory button too?

Reply:
Sure. Makes sense.

Reply:
Addressed